src/cloudai/schema/system/slurm/slurm_system.py

```
@@ -493,61 +493,99 @@ def scancel(self, job_id: int) -> None:
+        """
+        Parse the output from the 'sinfo' command to update node states.
+
+        Args:
+            sinfo_output (str): The output from the sinfo command.
+            node_user_map (dict): A dictionary mapping node names to users.
+        """
+        for line in sinfo_output.split("\n")[1:]:  # Skip the header line
             if not line.strip():
-                continue  # Skip empty lines
+                continue
             parts = line.split()
             partition, _, _, _, state, nodelist = parts[:6]
             partition = partition.rstrip("*")
-            node_names = self.parse_node_list([nodelist])
-
-            # Convert state to enum, handling states with suffixes
-            state_enum = self.convert_state_to_enum(state)
-
-            for node_name in node_names:
-                # Find the partition and node to update the state
-                for part_name, nodes in self.partitions.items():
-                    if part_name != partition:
-                        continue
-                    for node in nodes:
-                        if node.name == node_name:
-                            node.state = state_enum
-                            node.user = node_user_map.get(node_name, "N/A")
-                            break
+
+            node_groups = nodelist.split(",")
+            for node_group in node_groups:
+                node_names = self.parse_node_list([node_group.strip()])
+                state_enum = self.convert_state_to_enum(state)
+
+                for node_name in node_names:
+                    for part_name, nodes in self.partitions.items():
+                        if part_name != partition:
+                            continue
+                        for node in nodes:
+                            if node.name == node_name:
+                                node.state = state_enum
+                                node.user = node_user_map.get(node_name, "N/A")
```

Is it safe to have such value here?

---

Which value do you mean?

---

Do you mean "N/A"?

---

Yes, the `"N/A"`. Should we raise an issue right here or maybe on a level higher if that is non-recoverable issue? We probably want to raise issues as early as possible, otherwise it will be harder to debug. And harder for our users to understand the issue itself.

---

I understand this can be an issue and it can be difficult to debug, but I wanted to delay the decision because interfacing with Slurm is tricky. The node_user_map is responsible for maintaining a map between node names and user names. The idea is to use the squeue result to identify the user of a node. A corner case occurs when a node is allocated in the middle of execution and assigned to another user, making the node_user_map outdated. Do you think it's better to raise an exception right now rather than delaying the decision?

---

I think we need to have a discussion about error handling approach that we'd like to take. For now, IMO it is OK to have it like this.